include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -3556,6 +3556,16 @@ ORT_API_STATUS(OrtSessionOptionsAppendExecutionProvider_CUDA, _In_ OrtSessionOpt
 */
 ORT_API_STATUS(OrtSessionOptionsAppendExecutionProvider_MIGraphX, _In_ OrtSessionOptions* options, int device_id);
 
+/*
+ * This is the old way to add the ROCM provider to the session, please use
+ * SessionOptionsAppendExecutionProvider_ROCM above to access the latest functionality
+ * This function always exists, but will only succeed if Onnxruntime was built with
+ * HIP support and the ROCM provider shared library exists
+ *
+ * \param device_id HIP device id, starts from zero.
+*/
+ORT_API_STATUS(OrtSessionOptionsAppendExecutionProvider_ROCM, _In_ OrtSessionOptions* options, int device_id);
```

Comment:
If you don not really need to provide binary backward compatibility of this function, you may remove it to reduce our maintenance cost and reduce confusion to end users.  From what I know, neither CUDA EP nor ROCM EP has binary backward compatibility.  But it is up to you.